CHANGELOG.md

```diff
@@ -6,6 +6,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [9.0.0]
+### Changed
+- Bump dependency `semver` from `^5.7.1` to `^7.6.0` ([#181](https://github.com/MetaMask/utils/pull/181)).
+
+### Fixed
+- **BREAKING:** Replace dependency `superstruct` `^1.0.3` with ESM-compatible `@metamask/superstruct` `^3.0.0` ([#185](https://github.com/MetaMask/utils/pull/185)).
```

Comment:
Maybe we should note why this is breaking? Or _is_ this breaking? I think we still use the same functionality.

Reply:
Here's the [compare link](https://github.com/ianstormtaylor/superstruct/compare/v1.0.3...MetaMask:superstruct:v3.0.0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519) for the package.json diff between these versions. The `exports` field and the dual type declaration files are added after we switch to `@metamask/superstruct`.

Reply:
Or wait would this dependency replacement not be a breaking change even if the dependency contains breaking changes?

Reply:
My intuition is that since `superstruct@^1.0.3` broke downstream packages of `@metamask/utils` (e.g. `@metamask/{abi-utils,eth-snap-keyring,keyring-api,snaps-registry,snaps-sdk,snaps-utils}`) by preventing them from being imported into TypeScript projects that use the `NodeNext` setting for `moduleResolution`, the fixes introduced by `@metamask/superstruct` should count as breaking changes.

By that logic, it seems the type declaration files fix might also be a breaking change, since it fixes an issue that was affecting downstream packages, even via nested dependencies.

Both of these might count as cases where [a fix may need to be announced as a breaking change](https://github.com/MetaMask/core/blob/main/docs/reviewing-release-prs.md#fixes).

Reply:
Hmm, I don't know that I agree. `superstruct` is definitely unusable in a project that uses `moduleResolution` of `nodenext`, and that means that `@metamask/utils` is unusable too. So, upgrading this dependency makes `@metamask/utils` usable again (i.e. it's a fix). So, are there cases where this upgrade would break other kinds of TypeScript projects that currently use `@metamask/utils` in conjunction with other kinds of TS compiler settings? I can't think of anything but maybe I'm missing something?

Reply:
All good points! I'm finding that reverting the TypeScript version and `module`, `moduleResolution` settings on the [core v5.0 upgrade PR](https://github.com/MetaMask/core/pull/3645/) does cause breakage, but so far I'm mostly seeing errors related to the `multiformats` dynamic import. I'll try out other compiler options combinations to see if any errors related to `@metamask/utils` pop up.

Reply:
It doesn't look like the switch to `@metamask/superstruct` causes any new restrictions for downstream packages. Since there are no breaking changes left, I'll close this PR and create a new minor release.